chain/chain/src/chain.rs

```diff
@@ -972,11 +973,18 @@ impl Chain {
         // Get header we were syncing into.
         let header = self.get_block_header(&sync_hash)?;
         let hash = header.prev_hash;
-        let prev_header = self.get_block_header(&hash)?;
-        let tip = Tip::from_header(prev_header);
+        let prev_block = self.get_block(&hash)?;
+        let new_tail = prev_block.header.inner_lite.height;
+        let new_chunk_tail =
+            prev_block.chunks.iter().map(|x| x.inner.height_created).min().unwrap();
+        let tip = Tip::from_header(&prev_block.header);
         // Update related heads now.
         let mut chain_store_update = self.mut_store().store_update();
         chain_store_update.save_body_head(&tip)?;
+        // New Tail can not be earlier than `prev_block.header.inner_lite.height`
+        chain_store_update.update_tail(new_tail);
+        // New Chunk Tail can not be earlier than minimum of height_created in Block `prev_block`
+        chain_store_update.update_chunk_tail(new_chunk_tail);
```

Comment:
Wouldn't this cause some data that exist before state sync to be never garbage collected?